### PR TITLE
feat: add kitty keyboard options and settings

### PR DESCRIPTION
**Pull request by aymanbagabas**
This adds the necessary options to enable/disable the kitty keyboard protocol.

Needs: https://github.com/charmbracelet/bubbletea/pull/1080
Fixes: https://github.com/charmbracelet/bubbletea/issues/869

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module github.com/charmbracelet/bubbletea
 go 1.18
 
 require (
-	github.com/charmbracelet/x/ansi v0.1.4
+	github.com/charmbracelet/x/ansi v0.1.5-0.20240813204730-a29dab672bf2
```
Comment (caarlos0) [inline go.mod:6]: we should prob make another release of ansi

**Review by caarlos0 — APPROVED** [review]

File: kitty.go [hunk 2]
```diff
@@ -7,31 +7,82 @@ import (
...
+	return setKittyKeyboardFlagsMsg(0)
+}
+
+// requestKittyKeyboardFlagsMsg is a message to request the current Kitty keyboard
+// progressive enhancement protocol flags.
+type requestKittyKeyboardFlagsMsg struct{}
+
+// RequestKittyKeyboardFlags is a command to request the current Kitty keyboard
+// progressive enhancement protocol flags from the terminal.
+func RequestKittyKeyboardFlags() Msg {
```
Comment (aymanbagabas) [inline kitty.go:45]: Should this be to make it align more with current bubbletea commands?

```suggestion
func KittyKeyboardFlags() Msg {
```

CC/ @caarlos0 @meowgorithm @bashbunni

Comment (caarlos0) [reply kitty.go:45]: not sure... I think prefixing with `Request` is clearer...

Comment (aymanbagabas) [reply kitty.go]: I renamed it to `KittyKeyboard` to match other commands such as `WindowSize`. Similar to `WindowSize` command, `KittyKeyboard` queries the terminal and returns a `KittyKeyboardMsg` in `Update`.

**Review by caarlos0 — APPROVED** [review]

**Pull request merged**